image_tools_py/setup.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from setuptools import setup
+from setuptools.command.install_scripts import install_scripts
+
+package_name = 'image_tools_py'
+
+
+# This configuration deserves some explanation.  The goal is to generate a
+# test file called something like 'test_showimage_cam2image__rmw_fastrtps_.py'
+# that contains the test from test_showimage_cam2image.py.in, customized
+# for the rmw_implementation we want to test and for absolute paths of this
+# installation.  To do that, we override the 'install_scripts' stage of
+# setuptools, since that is the stage that knows the paths to the 'cam2image_py'
+# and 'showimage_py' executables.
+#
+# TODO(clalancette): One downside to what is being done below is that the output
+# file is being left in the source directory.  That's currently necessary
+# because the pytest invocation only looks there for test scripts.  It would
+# be nicer to generate the output file in the 'build' directory and then point
+# pytest at it, but I don't know how to do that.
+class custom_install_scripts(install_scripts):
+
+    def run(self):
+        rmw_implementations = ['rmw_fastrtps_cpp']
```

Comment:
@clalancette do we need to generate a test in the first place? what about parametrizing the test itself? Also, IIRC `rmw_fastrtps_cpp` is the default RMW implementation so it shouldn't be necessary to even pull this off.

Reply:
There are two pieces to this answer.  We do need to generate *something*; we need to fill in the runtime paths to the `showimage_py` and `cam2image_py` executables.  While we don't need to parameterize around the RMW implementation, this is what the image_tools CMakeLists.txt [does](https://github.com/ros2/demos/blob/master/image_tools/CMakeLists.txt#L86), so I thought I would replicate it here.  But I don't feel strongly about it either way.

Reply:
Ok, fair enough.